### DISCUSSION TITLE
$if statements don't work with const

### DISCUSSION THREAD

**Discussion by radare** [vlang/v#11672] [Ideas]
**V version:** V 0.1.27 28ffe2a.7e538d7
**OS:** macOS 10.15

**What did you do?**
```
const (
	$if macos {
		a = 1
	} $else {
		a = 2
	}
)

fn main() {
	eprintln('$a')
}
```

**What did you expect to see?**

```
macos$ 1
```
 
**What did you see instead?**

```
test.v:2:2: error: unexpected `$`, expecting `name`
    1 | const (
    2 |     $if macos {
      |     ^
    3 |         a = 1
    4 |     } $else {
```

**Comment by Gladear** [comment]
If you need a workaround : put the const in backend specific files, ending with `_darwin`, `_unix`, etc. 🙂

**Comment by spytheman** [comment]
$if does not work outside of functions. It is not a bug, just not implemented yet.
